src/Pages/WatchLater/WatchLater.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState} from 'react';
+import { VideoCard } from '../../Shared';
+import './WatchLater.css'; 
+import { getWatchlaterVideos } from '../../Helpers/Services/actions';
+import { v4 as uuid } from "uuid";
+import { useAlert, useWatchlist } from '../../Context';
+
+const WatchLater = () => {
+
+    const [ videoData, setVideoData] = useState([]);
+    const {alertContent , setAlertContent} = useAlert();
+    const {watchLaterVideos} = useWatchlist();
+ 
+
+    const getAPI = (func, str) => {
+        func
+          .then((response) => {
+            if (response.data && response.status === 200) {
+                if (str === "getWatchlaterVideos") {
+                setVideoData(response?.data?.watchlater);
+              }
+            } else{
+                setAlertContent({_id: uuid(), isShow:true, type:'ERROR', content:"Unexpected error.Please try again later."})
+            }
+          })
+          .catch((error) => {
+            setAlertContent({_id: uuid(), isShow:true, type:'ERROR', content:"Unexpected error.Please try again later."})
+          });
+      };
+
+    useEffect(() => {
+        getAPI(getWatchlaterVideos(), "getWatchlaterVideos");
+      }, [watchLaterVideos]);
+
+
+    return(
+        <>
+            <p className='heading-md'>Watchlater...</p> 
+            <div className="col-12 flex-container-row watchlater-videos">
+    
+                {videoData?.length?
```

Comment:
why name this cIndex and cVal ? i think the variable naming could be better unless you have a good reason for the c.